service-workers/service-worker/resources/test-helpers.sub.js

```diff
@@ -58,6 +58,25 @@ function with_iframe(url) {
     });
 }
 
+function with_sandboxed_iframe(url, sandbox) {
```

Comment:
Yes, I added sandbox property and srcdoc property to the options as suggested. Looks cleaner.